### DISCUSSION TITLE
Increasing maxDuration for Nuxt 3 deployment

### DISCUSSION THREAD

**Discussion by mulhoon** [vercel/community#4804] [Help]
### Summary

I'd like to increase the `maxDuration` for endpoints in Nuxt 3's `server/api/` directory, but I can't target it with `vercel.json`.

### Steps to Reproduce

With Nuxt 3, all server/API endpoints are bundled under a single function called `__nitro`. So I guess I need to increase the `maxDuration` for this. Here is what the output of the nuxt build looks like:

<img width="247" alt="image" src="https://github.com/vercel/community/assets/1137909/2dfd7ed4-245d-47ef-8d8a-a0fd6d6e6c94">

But when I try to target this in `vercel.json` it doesn't work
```json
{
  "functions": {
    "api/__nitro/**/*": {
      "maxDuration": 300
    }
  }
}
```
I've tried all kinds of ways of writing it, and all fail with the same error:
```
Error: The pattern "api/__nitro/**/*" defined in `functions` doesn't match any Serverless Functions inside the `api` directory.
```
I also have this is my `nuxt.config.js`
```
nitro: {
   preset: 'vercel'
},
```
Does anyone know how to increase the `maxDuration` of a Nuxt 3 build on Vercel?
Many Thanks

**Comment by kstraszewski** [comment]
Hi You need to add this to nuxt.config.js -> nitro: 

```ts
// nuxt.config.ts
  nitro: {
    vercel: {
      functions: {
        maxDuration: 300,
      }
    }
  }

```

**Reply by grokpot** [reply]
Hi @kstraszewski and @mulhoon - THANK YOU! This was a hard solution to find.
A couple followup questions for @kstraszewski :
1. Where is it documented that vercel.json config can go in nitro vercel nuxt config?
2. How would the solution look if it was solved using vercel.json?

Thanks!

**Reply by kstraszewski** [reply]
Hi unfortunately I have no idea how it could look, when you will use the vercel.json. However, I found this solution by scrolling through typescript types in nuxt.config > nitro > vercel. All of those are well typed.
